localstorage.js

```diff
@@ -0,0 +1,64 @@
+// JS for syncing form inputs to local storage (iff enabled)
+
+// defines the HTML IDs of every input that should be saved here.
+// Code is set up to handle anything with `value` and bootstrap checkboxes right now
+const formsOfInterest = ['doh-url', 'cors-switch', 'doh-method', 'doh-qtype', 'dnssec-switch'];
+
+// we have to save references to listeners here so they can be removed later
+const listenerDict = {};
+const saveSwitch = document.getElementById('save-switch');
+
+document.addEventListener('DOMContentLoaded', function(e) {
+    saveSwitch.addEventListener("change", handleSaveSwitch);
+    // try loading everything. If anything loads, set the save switch
+    formsOfInterest.forEach((id) => {
+        const elem = document.getElementById(id);
+        const val = localStorage.getItem(id);
+        if (val !== null ) {
+            elem.type === "checkbox" ? elem.checked = val : elem.value = val;
```

Comment:
We should change this so `val` is properly converted to boolean.
Alternatively, we could treat them like environment variables (i.e. [here](https://github.com/byu-imaal/dohjs/pull/45/files#diff-6dc4294a01cc7cb81592c8be9d96130dR62) we save `false` as `''` and `true` as anything that's not empty string)